src/Symfony/Component/Console/Tests/Helper/TableTest.php

```diff
@@ -974,6 +974,78 @@ public function testGetStyleDefinition()
         Table::getStyleDefinition('absent');
     }
 
+    /**
+     * @dataProvider renderSetTitle
+     */
+    public function testSetTitle($title, $style, $expected)
+    {
+        (new Table($output = $this->getOutputStream()))
+            ->setTitle($title)
+            ->setHeaders(array('ISBN', 'Title', 'Author'))
+            ->setRows(array(
+                array('99921-58-10-7', 'Divine Comedy', 'Dante Alighieri'),
+                array('9971-5-0210-0', 'A Tale of Two Cities', 'Charles Dickens'),
+                array('960-425-059-0', 'The Lord of the Rings', 'J. R. R. Tolkien'),
+                array('80-902734-1-6', 'And Then There Were None', 'Agatha Christie'),
+            ))
+            ->setStyle($style)
+            ->render()
+        ;
+
+        $this->assertEquals($expected, $this->getOutputContent($output));
+    }
+
+    public function renderSetTitle()
+    {
+        return array(
+            array(
+                'Books',
+                'default',
+                <<<'TABLE'
++---------------+----------- Books --------+------------------+
+| ISBN          | Title                    | Author           |
++---------------+--------------------------+------------------+
+| 99921-58-10-7 | Divine Comedy            | Dante Alighieri  |
+| 9971-5-0210-0 | A Tale of Two Cities     | Charles Dickens  |
+| 960-425-059-0 | The Lord of the Rings    | J. R. R. Tolkien |
+| 80-902734-1-6 | And Then There Were None | Agatha Christie  |
++---------------+--------------------------+------------------+
+
+TABLE
+            ),
+            array(
+                'Books',
+                'box',
+                <<<'TABLE'
+┌───────────────┬─────────── Books ────────┬──────────────────┐
+│ ISBN          │ Title                    │ Author           │
+├───────────────┼──────────────────────────┼──────────────────┤
+│ 99921-58-10-7 │ Divine Comedy            │ Dante Alighieri  │
+│ 9971-5-0210-0 │ A Tale of Two Cities     │ Charles Dickens  │
+│ 960-425-059-0 │ The Lord of the Rings    │ J. R. R. Tolkien │
+│ 80-902734-1-6 │ And Then There Were None │ Agatha Christie  │
+└───────────────┴──────────────────────────┴──────────────────┘
+
+TABLE
+            ),
+            array(
+                'Boooooooooooooooooooooooooooooooooooooooooooooooooooooooooks',
+                'default',
+                <<<'TABLE'
++ Booooooooooooooooooooooooooooooooooooooooooooooooooooooo... +
```

Comment:
What if we leave at least 1 separator character for long titles in the head? Instead of this:

```
+ Booooooooooooooooooooooooooooooooooooooooooooooooooooooo... +
| ISBN          | Title                    | Author           |
+---------------+--------------------------+------------------+
| 99921-58-10-7 | Divine Comedy            | Dante Alighieri  |
| ...
```

This:

```
+- Booooooooooooooooooooooooooooooooooooooooooooooooooooo... -+
| ISBN          | Title                    | Author           |
+---------------+--------------------------+------------------+
| 99921-58-10-7 | Divine Comedy            | Dante Alighieri  |
| ...
```

Reply:
Done. That looks much better!